backstory/src/main/java/com/google/sps/data/PerspectiveAnalysis.java

```diff
@@ -0,0 +1,104 @@
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import au.com.origma.perspectiveapi.v1alpha1.PerspectiveAPI;
+import au.com.origma.perspectiveapi.v1alpha1.models.AnalyzeCommentRequest;
+import au.com.origma.perspectiveapi.v1alpha1.models.AnalyzeCommentRequest.Builder;
+import au.com.origma.perspectiveapi.v1alpha1.models.AnalyzeCommentResponse;
+import au.com.origma.perspectiveapi.v1alpha1.models.AttributeScore;
+import au.com.origma.perspectiveapi.v1alpha1.models.AttributeType;
+import au.com.origma.perspectiveapi.v1alpha1.models.ContentType;
+import au.com.origma.perspectiveapi.v1alpha1.models.Entry;
+import au.com.origma.perspectiveapi.v1alpha1.models.RequestedAttribute;
+import au.com.origma.perspectiveapi.v1alpha1.models.Score;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class PerspectiveAnalysis {
+  /** a private class that keeps the analysis and type for each AttributeType we want analyzed */
+  private class AttributeAnalysis {
+    /** the attribute score given by analysis */
+    private final float score;
+    /** the type that the analysis score is for */
+    private final AttributeType type;
+
+    public AttributeAnalysis(float score, AttributeType type) {
+      this.score = score;
+      this.type = type;
+    }
+  }
+
+  /** an array of all the types we want analysis on */
+  private static final AttributeType[] types = {AttributeType.ATTACK_ON_AUTHOR,
+      AttributeType.ATTACK_ON_COMMENTER, AttributeType.FLIRTATION, AttributeType.IDENTITY_ATTACK,
+      AttributeType.INCOHERENT, AttributeType.INSULT, AttributeType.LIKELY_TO_REJECT,
+      AttributeType.OBSCENE, AttributeType.PROFANITY, AttributeType.SEVERE_TOXICITY,
+      AttributeType.SEXUALLY_EXPLICIT, AttributeType.SPAM, AttributeType.THREAT,
+      AttributeType.UNSUBSTANTIAL};
+
+  /** the text that generated these scores */
+  private final String text;
+
+  /** the scores & their types from the analysis of the text */
+  private final AttributeAnalysis[] analyses;
+
+  public PerspectiveAnalysis(PerspectiveAPI perspective, String text) throws NullPointerException {
+    this.text = text;
+
+    AnalyzeCommentRequest.Builder builder = new AnalyzeCommentRequest.Builder().comment(
+        new Entry.Builder().type(ContentType.PLAIN_TEXT).text(text).build());
+
+    // add all the types we want to this builder
+    for (AttributeType type : types) {
+      builder.addRequestedAttribute(type, null);
+    }
+
+    AnalyzeCommentRequest request = builder.build();
+    AnalyzeCommentResponse response = perspective.analyze(request);
```

Comment:
I like how you created a value container for analysis data. One comment though is that we typically don't see a value-container-type object making network calls, and doing this from the constructor can be especially problematic... for example, if the request fails, would this constructor return an invalid object?

More conventional would be to have a service client:

```
class PerspectiveServiceClient {
  PerspectiveAnalysis analyze() {
    // Make request.
    AnalyzeCommentResponse response = perspective.analyze(request);
    // Convert response to PerspectiveAnalysis.
    return analysis;
  }
}

```

Reply:
I actually move this code into the Manager class later. Will that solve the issue or should I still write a service client?

Reply:
Done